app/routes.py

```diff
@@ -1,2 +1,346 @@
+def check_in_video():
+    """Checks a customer's video back in"""
+    request_body = request.get_json() 
+
+    for detail in ["customer_id", "video_id"]: 
+        if not (detail in request_body):
+            return make_response({}, 400)
+    returned_video = Rental(customer_id=request_body["customer_id"],
+                            video_id=request_body["video_id"])
+    customer = Customer.query.get(returned_video.customer_id)
+    video = Video.query.get(returned_video.video_id)
+
+    if customer is None or video is None:
+        return make_response({"details": "The customer and/or the video does not exist."}, 404)
+    if (video.video_id != request_body["video_id"]) or (customer.customer_id != request_body["customer_id"]):
+        return make_response("", 400)
+    if video.available_inventory == video.total_inventory:
+        return make_response({
+            "customer_id": customer.customer_id,
+            "video_id": video.video_id,
+            "videos_checked_out_count": customer.videos_checked_out_count, 
+            "available_inventory": video.available_inventory
+            }, 400)
+
+    customer.videos_checked_out_count -= 1 
+    video.available_inventory += 1
+
+    db.session.add(returned_video)
+    db.session.commit()
+    return jsonify({
+            "customer_id": customer.customer_id,
+            "video_id": video.video_id,
+            "videos_checked_out_count": customer.videos_checked_out_count, 
+            "available_inventory": video.available_inventory
+            })
+
+@customer_bp.route("/<customer_id>/rentals", methods=["GET"])
+def list_customer_videos(customer_id):
+    """Retrieves all videos a customer has checked out"""
+    hold_whats_relevant = {} 
+    hold_dictified_entities = [] 
+    customer_entity = [] 
+    dictified_ce = {} 
+    video_entity = [] 
+    dictified_ve = {}
+    rental_entity = []
+    dictified_re = {}
+
+    customer_vids = db.session.query(Customer, Video, Rental).join(Customer, Customer.customer_id==Rental.customer_id)\
+        .join(Video, Video.video_id==Rental.video_id).filter(Customer.customer_id==customer_id).all()
+
+    for tupled_entity in customer_vids: 
+        if tupled_entity[0] == None:
+            return make_response({"details": "Customer does not exist"}, 404)
+        for entity in tupled_entity: 
+            if type(entity) == Customer: 
+                if entity.videos_checked_out_count == 0:
+                    return jsonify([])
+                customer_entity.append(entity)
+                dictified_ce["id"] = entity.customer_id
+                dictified_ce["name"] = entity.name
+                dictified_ce["phone"] = entity.phone_number
+                dictified_ce["postal_code"] = entity.postal_code
+                dictified_ce["registered_at"] = entity.register_at
+                dictified_ce["videos_checked_out_count"] = entity.videos_checked_out_count
+                hold_dictified_entities.append(dictified_ce)
```

Comment:
You can use `to_json` instead of producing this dictionary manually (the fields match up):

```suggestion
                customer_entity.append(entity)
                hold_dictified_entities.append(entity.to_json())
```